### PR TITLE
feat: add revocation support

### LINKED ISSUE #204: Support revocation
We should look at supporting revocation in the cloud api. This is mostly already handled by ACA-Py so we should just look at what is needed in terms of infrastructure and API endpoints. 

- [ ] Add tails server to the infrastructure and provide this to all ACA-Py agents
- [ ] Allow to specify `supports_revocation` when creating a credential definition
- [ ] Add endpoint to revoke a credential by the credential exchange record id or the indy specific identifiers.
- [ ] add tests

This is not a requirement for the V1 Must Have

### PR DESCRIPTION
**Pull request by morrieinmaas**
closes #204

### CODE REVIEW

**Comment by morrieinmaas** [comment]
requires merge of #261 related PR

File: app/generic/definitions.py [hunk 1]
```diff
@@ -248,11 +257,69 @@ async def create_credential_definition(
...
+                            },
+                            timeout=30,
+                        )
+                    except TimeoutError:
+                        raise CloudApiException(
+                            "Failed to retrieve transaction record for endorser", 500
+                        )
+                    finally:
+                        await stop_listener()
+
```
Comment (TimoGlastra) [inline app/generic/definitions.py:311]: starting the listener after the actions that triggers the event can possibly lead to race conditions

File: app/generic/definitions.py [hunk 2]
```diff
@@ -248,11 +257,69 @@ async def create_credential_definition(
             raise CloudApiException(
                 "Unable to construct credential definition id from signature response"
             ) from e
-
     else:
-        await stop_listener()
         credential_definition_id = result.credential_definition_id
 
+    if credential_definition.support_revocation:
```
Comment (TimoGlastra) [inline app/generic/definitions.py:263]: Does this need to be done manually? I know they autmoated 99 percent of it, and thought this was also handled, but not 100% sure

Comment (morrieinmaas) [reply app/generic/definitions.py:263]: Hmm let me double check this, but I didn't get this to work otherwise

**Comment by morrieinmaas** [comment]
@TimoGlastra thx for the feedback already. I still need to add some tests and e2e tests for the revoking a credential. So it's WIP. Just added it and manually verified it uses revocation registry

File: app/generic/definitions.py (hunk 2 above)
Comment (morrieinmaas) [reply app/generic/definitions.py:263]: so not doing this leaves credential issuance in 'abandoned' state. 

I'll get to the bottom of it

**Comment by lohanspies** [comment]
> @morrieinmaas will take a look at this one when all other PRs are merged and this one is up to date with main

@TimoGlastra can you perhaps review and close this PR?

**Pull request merged**